### PR TITLE
test(gui): add vfs tests

### PR DESCRIPTION
**Pull request by saw-jan**
Add vfs tests to hydrate and dehydrate files and folders:
Test cases:
- hydrate file by opening it
- hydrate file (file explorer action: `Always keep on this device`)
- dehydrate file (file explorer action: `Free up space`)
- hydrate folder (file explorer option)
- add files/folders inside hydrated folder locally
- add files/folders inside hydrated folder from server
- add files/folders inside dehydrated folder locally
- add files/folders inside dehydrated folder from server
- dehydrate folder (file explorer action)

### CODE REVIEW

File: test/gui/tst_vfs/test.feature [hunk 1]
```diff
@@ -57,3 +57,118 @@ Feature: VFS support
...
+        And user "Alice" has uploaded file with content "some contents" to "nested/subfol1/subfol2/subfol3/subfile3.txt" in the server
+        And user "Alice" has uploaded file with content "some contents" to "nested/subfol1/subfol2/subfol3/subfol4/subfile4.txt" in the server
+        And user "Alice" has set up a client with default settings
+        Then the placeholder file "simple.txt" should exist on the file system
+        And the placeholder file "nested/lorem.txt" should exist on the file system
+        And the placeholder file "nested/subfol1/subfol2/subfol3/subfol4/subfile4.txt" should exist on the file system
+
+        # mark sub folder as available-locally
+        When user "Alice" reads the content of file "nested/subfol1/subfol2/subfile2.txt"
+        And user "Alice" marks folder "nested/subfol1" as available-locally from the file explorer
```
Comment (anon-pradip) [inline test/gui/tst_vfs/test.feature]: Windows shows "Always keep on this device" option, so better we use that same phrase instead of "available-locally", for better readability? 🤔

File: test/gui/tst_vfs/test.feature [hunk 2]
```diff
@@ -57,3 +57,118 @@ Feature: VFS support
...
+        When user "Alice" marks file "simple.txt" as available-locally from the file explorer
+        And the user waits for file "simple.txt" to be synced
+        Then the file "simple.txt" should be downloaded
+        And the placeholder file "large.txt" should exist on the file system
+
+        # mark files online-only
+        When user "Alice" marks file "testFile.txt" as online-only from the file explorer
+        And the user waits for file "testFile.txt" to be synced
+        Then the placeholder file "testFile.txt" should exist on the file system
+        When user "Alice" marks file "parent/lorem.txt" as online-only from the file explorer
```
Comment (anon-pradip) [inline test/gui/tst_vfs/test.feature]: Windows file explorer's context menu shows "Free up space" for making file(s) available online only. So better we use that same phrase here also instead of "online-only", for better readability ? 🤔

File: test/gui/tst_vfs/test.feature [hunk 3]
```diff
@@ -57,3 +57,118 @@ Feature: VFS support
...
+        And the placeholder file "large.txt" should exist on the file system
+        And the placeholder file "parent/lorem.txt" should exist on the file system
+        And the placeholder file "parent/epsum.txt" should exist on the file system
+
+        # Hydrate some files by reading the content
+        When user "Alice" reads the content of file "testFile.txt"
+        And user "Alice" reads the content of file "parent/lorem.txt"
+        Then the file "testFile.txt" should be downloaded
+        And the file "parent/lorem.txt" should be downloaded
+        And the placeholder file "parent/epsum.txt" should exist on the file system
```
Comment (anon-pradip) [inline test/gui/tst_vfs/test.feature:82]: Confused if we need to check:
`And the placeholder file "parent/epsum.txt" should exist on the file system` because we have not touched this in "# Hydrate some files by reading the content" and also we already checked placeholder of file `parent/epsum.txt` in L75
?

Comment (saw-jan) [reply test/gui/tst_vfs/test.feature:82]: this is to check that the action is one file doesn't affect others. kind of edge case so kept the checks to minimal
